amgut/lib/data_access/ag_data_access.py

```diff
@@ -540,31 +564,33 @@ def ag_verify_kit_password_change_code(self, email, kitid, passcode):
         kitid is the supplied_kit_id in the ag_kit table
         passcode is the password change verification value
         """
-        cursor = self.get_cursor()
-        cursor.callproc('ag_verify_password_change_code', [email, kitid,
-                                                           passcode])
-        return cursor.fetchone()[0]
-
-    def getBarcodesByKit(self, kitID):
+        sql = """SELECT EXISTS(SELECT pass_reset_time
+                 FROM ag_kit
+                 INNER JOIN ag_login USING (ag_login_id)
+                 WHERE pass_reset_code = %s and email = %s
+                 AND supplied_kit_id = %s
+                 AND current_time_ < pass_reset_time)"""
+        return self._sql.execute_fetchone(sql, [passcode, email, kitid])[0]
+
+    def getBarcodesByKit(self, kitid):
         """Returns a list of barcodes in a kit
 
-        kitID is the supplied_kit_id from the ag_kit table
+        kitid is the supplied_kit_id from the ag_kit table
         """
-        results = self._sql.execute_proc_return_cursor(
-            'ag_get_barcodes_by_kit', [kitID])
-        barcodes = [row[0] for row in results]
-        results.close()
-        return barcodes
+        sql = """SELECT barcode
+                 FROM ag_kit_barcodes
+                 INNER JOIN ag_kit kUSING(ag_kit_id)
```

Comment:
bug that would bomb the query

this implies that this PR hasn't been exercised... can you verify locally please?